content/articles/nextjs-dnd-file-upload/index.md

```diff
@@ -0,0 +1,427 @@
+  const handleDragOver = (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+
+    // set dropEffect to copy i.e copy of the source item
+    e.dataTransfer.dropEffect = "copy";
+    dispatch({ type: "SET_IN_DROP_ZONE", inDropZone: true });
+  };
+
+  // onDrop sets inDropZone to false and adds files to fileList
+  const handleDrop = (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+
+    // get files from event on the dataTransfer object as an array
+    let files = [...e.dataTransfer.files];
+
+    // ensure a file or files are dropped
+    if (files && files.length > 0) {
+      // loop over existing files
+      const existingFiles = data.fileList.map((f) => f.name);
+      // check if file already exists, if so, don't add to fileList
+      // this is to prevent duplicates
+      files = files.filter((f) => !existingFiles.includes(f.name));
+
+      // dispatch action to add droped file or files to fileList
+      dispatch({ type: "ADD_FILE_TO_LIST", files });
+      // reset inDropZone to false
+      dispatch({ type: "SET_IN_DROP_ZONE", inDropZone: false });
+    }
+  };
+
+  return (
+    <>
+      <div
+        className={styles.dropzone}
+        onDragEnter={(e) => handleDragEnter(e)}
+        onDragOver={(e) => handleDragOver(e)}
+        onDragLeave={(e) => handleDragLeave(e)}
+        onDrop={(e) => handleDrop(e)}
+      >
+        <Image src="/upload.svg" alt="upload" height={50} width={50} />
+
+        <input
+          id="fileSelect"
+          type="file"
+          multiple
+          className={styles.files}
+        />
+        <label htmlFor="fileSelect">You can select multiple Files</label>
+
+        <h3 className={styles.uploadMessage}>
+          or drag &amp; drop your files here
+        </h3>
+      </div>
+      {/* Pass the selectect or dropped files as props */}
+      <FilePreview fileData={data} />
+    </>
+  );
+};
+
+export default DropZone;
+```
+
+This will turn the `DropZone.js` component into a dropzone region. The `handleDragEnter`, `dragenter`, `handleDragLeave`, `handleDragOver` and the `handleDrop` function will be used to prevent the default behaviour, i.e event propagation from child to parent. 
+
+The `handleDrop` function will be used to prevent the default behaviour which is to open the file on the browser. It instead lets you define the custom behaviour to handle the file drop.
+
+`onDragEnter` and `onDragOver` set the `inDropZone` state to true since the user is dragging a file over the valid dropzone region. `onDragLeave` set the `inDropZone` state to false since the user is dragging a file away from the valid dropzone region.
+
+`onDrop` will set the `inDropZone` to get the fileList from the event on the dataTransfer object as an array, iterate over existing files, check if the file already exists, if so, don't add to fileList, this is to prevent duplicates. Then dispatch an action to add dropped files or files to fileList. Finally, reset `inDropZone` to false.
```

Comment:
The first sentence here is too long, kindly break it down into constructive sentences. Also, the second sentence doesn't flow well.